google/cloud/security/inventory/pipeline_builder.py

```diff
@@ -0,0 +1,241 @@
+            {'module_name': 'load_projects_buckets_acls_pipeline',
+             'depends_on': 'buckets',
+             'api_name': 'gcs_api',
+             'dao_name': 'bucket_dao'},
+        'cloudsql': 
+            {'module_name': 'load_projects_cloudsql_pipeline',
+             'depends_on': 'projects',
+             'api_name': 'cloudsql_api',
+             'dao_name': 'cloudsql_dao'},
+        'firewall_rules': 
+            {'module_name': 'load_firewall_rules_pipeline',
+             'depends_on': 'projects',
+             'api_name': 'compute_beta_api',
+             'dao_name': 'project_dao'},
+        'folders':
+            {'module_name': 'load_folders_pipeline',
+             'depends_on': 'organizations',
+             'api_name': 'crm_v2beta1_api',
+             'dao_name': 'dao'},
+        'forwarding_rules': 
+            {'module_name': 'load_forwarding_rules_pipeline',
+             'depends_on': 'projects',
+             'api_name': 'compute_api',
+             'dao_name': 'forwarding_rules_dao'},
+        'group_members':
+            {'module_name': 'load_group_members_pipeline',
+             'depends_on': 'groups',
+             'api_name': 'admin_api',
+             'dao_name': 'dao'},
+        'groups':
+            {'module_name': 'load_groups_pipeline',
+             'depends_on': 'organizations',
+             'api_name': 'admin_api',
+             'dao_name': 'dao'},
+        'org_iam_policies':
+            {'module_name': 'load_org_iam_policies_pipeline',
+             'depends_on': 'organizations',
+             'api_name': 'crm_api',
+             'dao_name': 'organization_dao'},
+        'organizations': 
+            {'module_name': 'load_orgs_pipeline',
+             'depends_on': None,
+             'api_name': 'crm_api',
+             'dao_name': 'organization_dao'},
+        'projects':
+            {'module_name': 'load_projects_pipeline',
+             'depends_on': 'organizations',
+             'api_name': 'crm_api',
+             'dao_name': 'project_dao'},
+        'projects_iam_policies':
+            {'module_name': 'load_projects_iam_policies_pipeline',
+             'depends_on': 'projects',
+             'api_name': 'crm_api',
+             'dao_name': 'project_dao'},
+    }
+
+    def __init__(self, cycle_timestamp, pipeline_configs, flag_configs,
+                 api_map, dao_map):
+        """Initialize the pipeline builder.
+
+        Args:
+            cycle_timestamp: String of timestamp, formatted as YYYYMMDDTHHMMSSZ.
+            pipeline_configs: String of the path to the pipeline configs file.
+            flag_configs: Dictionary of flag values.
+            api_map: Dictionary of GCP API instances, mapped to each resource.
+            dao_map: Dictionary of DAO instances, mapped to each resource.
+
+        Returns:
+            None
+        """
+        self.cycle_timestamp = cycle_timestamp
+        self.pipeline_configs = file_loader.read_and_parse_file(
```

Comment:
nit: there's technically only one 'config' the plural isn't accurate. the use of pipeline is also somewhat redundant.

s/pipeline_configs/config/
s/flag_configs/flags/